consensus/clique/src/main/java/tech/pegasys/pantheon/consensus/clique/CliqueProtocolSchedule.java

```diff
@@ -27,44 +35,33 @@
   public static ProtocolSchedule<CliqueContext> create(
       final GenesisConfigOptions config, final KeyPair nodeKeys) {
 
-    // Get Config Data
     final CliqueConfigOptions cliqueConfig = config.getCliqueConfigOptions();
-    final long epochLength = cliqueConfig.getEpochLength();
-    final long blockPeriod = cliqueConfig.getBlockPeriodSeconds();
-    final int chainId = config.getChainId().orElse(DEFAULT_CHAIN_ID);
-
-    final MutableProtocolSchedule<CliqueContext> protocolSchedule =
-        new MutableProtocolSchedule<>(chainId);
 
     // TODO(tmm) replace address with passed in node data (coming later)
-    final CliqueProtocolSpecs specs =
-        new CliqueProtocolSpecs(
-            blockPeriod,
-            epochLength,
-            Util.publicKeyToAddress(nodeKeys.getPublicKey()),
-            protocolSchedule);
+    final Address localNodeAddress = Util.publicKeyToAddress(nodeKeys.getPublicKey());
 
-    protocolSchedule.putMilestone(0, specs.frontier());
-
-    config
-        .getHomesteadBlockNumber()
-        .ifPresent(blockNumber -> protocolSchedule.putMilestone(blockNumber, specs.homestead()));
-    config
-        .getTangerineWhistleBlockNumber()
-        .ifPresent(
-            blockNumber -> protocolSchedule.putMilestone(blockNumber, specs.tangerineWhistle()));
-    config
-        .getSpuriousDragonBlockNumber()
-        .ifPresent(
-            blockNumber -> protocolSchedule.putMilestone(blockNumber, specs.spuriousDragon()));
-    config
-        .getByzantiumBlockNumber()
-        .ifPresent(blockNumber -> protocolSchedule.putMilestone(blockNumber, specs.byzantium()));
-    config
-        .getConstantinopleBlockNumber()
-        .ifPresent(
-            blockNumber -> protocolSchedule.putMilestone(blockNumber, specs.constantinople()));
+    final EpochManager epochManager = new EpochManager(cliqueConfig.getEpochLength());
+    return new ProtocolScheduleBuilder<>(
+            config,
+            builder ->
+                applyCliqueSpecificModifications(
+                    epochManager, cliqueConfig.getBlockPeriodSeconds(), localNodeAddress, builder))
+        .createProtocolSchedule(DEFAULT_CHAIN_ID);
```

Comment:
Passing in a magic default is a bit of smell - reads as though we are creating a protocol schedule of the DEFAULT_CHAIN_ID (rather than this being an override).
Also - can't quite appreciate the difference between passing everything in at construction vs in the call to createProtocolSchedule.

Reply:
Yeah our handling of default chain ID is definitely weird, but not new here (I think it used to be a bit better hidden).  I suspect we should default to no chain ID if one isn't in the config file which would clean this up but that probably requires some additional research and thought.

I have changed it so that the default chain ID is passed into the constructor instead of the createProtocolSchedule method to be more consistent.